### PR TITLE
Better document scenarios for log vs. notify_group

### PR DESCRIPTION
**Pull request by tas50**
I ran into this one and had a head scratcher over how to fix it.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/rubocop/cop/chef/deprecation/log_resource_notifications.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ module RuboCop
   module Cop
     module Chef
       module Deprecations
-        # In Chef Infra Client 16 the log resource no longer notifies when logging so notifications should not be triggered from log resources. Use the notify_group resource introduced in Chef Infra Client 15.8 instead to aggregate notifications.
+        # In Chef Infra Client 16 and later the log resource no longer notifies when logging so notifications should not be triggered from log resources. Use the notify_group resource introduced in Chef Infra Client 15.8 instead to aggregate notifications.
```
Comment (kagarmoe) [inline lib/rubocop/cop/chef/deprecation/log_resource_notifications.rb:22]: Notifications triggered by the log resource are deprecated beginning with Chef Infra Client 16 (April 2020). The log resource no longer notifies when logging and cannot be used successfully to trigger notifications from the log resource. Use the notify_group resource introduced in Chef Infra Client 15.8 for notifications.

I don't know how important `aggregate` is for describing this.

**Review by kagarmoe — COMMENTED** [review]
A suggestion. The simplification may obscure more than it clarifies.

File: lib/rubocop/cop/chef/deprecation/log_resource_notifications.rb (hunk 1 above)
Comment (tas50) [reply lib/rubocop/cop/chef/deprecation/log_resource_notifications.rb:22]: it's not deprecated in 16. it's just gone

File: lib/rubocop/cop/chef/deprecation/log_resource_notifications.rb [hunk 2]
```diff
@@ -43,6 +47,12 @@ module Deprecations
         #     notifies :restart, 'service[foo]', :delayed
         #   end
         #
+        #   notify_group 'a notify_group resource that is never notified but just fires'
```
Comment (jaymzh) [inline lib/rubocop/cop/chef/deprecation/log_resource_notifications.rb:50]: noice

**Review by jaymzh — APPROVED** [review]

**Pull request closed without merging**